crates/store/src/db/mod.rs

```diff
@@ -187,21 +184,29 @@ impl From<NoteRecord> for NoteSyncRecord {
 }
 
 impl Db {
-    /// Open a connection to the DB, apply any pending migrations, and ensure that the genesis block
-    /// is as expected and present in the database.
-    // TODO: This span is logged in a root span, we should connect it to the parent one.
-    #[instrument(target = COMPONENT, skip_all)]
-    pub async fn setup(
-        database_filepath: PathBuf,
-        genesis_filepath: &str,
-        block_store: Arc<BlockStore>,
-    ) -> Result<Self, DatabaseSetupError> {
-        info!(target: COMPONENT, ?database_filepath, "Connecting to the database");
-
-        if let Some(p) = database_filepath.parent() {
-            create_dir_all(p).map_err(DatabaseError::IoError)?;
-        }
+    pub fn bootstrap(database_filepath: PathBuf, genesis: &GenesisBlock) -> anyhow::Result<()> {
```

Comment:
I would add brief doc comments here.

Also, do we not want to instrument this function?

Reply:
I'm on the fence tbh. Mostly because this is a once-off so its not like there are statistics or metrics we can glean from this.

On the other hand there isn't really a large downside besides polluting the logs/traces a tiny bit.

Reply:
That said, we probably _do_ want to instrument the database migrations. So maybe we should just do it :)

Reply:
Something that could be nice is actually using these traces during development. You can have a separate backend environment per developer, and they can use this to test/inspect/debug slowness and flows etc.